frontend/src/scenes/plugins/pluginsLogic.ts

```diff
@@ -34,6 +34,17 @@ function capturePluginEvent(event: string, plugin: PluginType, type?: PluginInst
     })
 }
 
+async function loadPaginatedResults(url: string | null, maxIterations: number = 10): Promise<any[]> {
+    let results: any[] = []
+    let i = 0
+    while (url && ++i <= maxIterations) {
+        const { results: partialResults, next } = await api.get(url)
+        results = results.concat(partialResults)
+        url = next
+    }
```

Comment:
for loops with exiting out are easier to read in cases like this imho (easier to see it does what I expect), sth like this  not sure on the ts syntax
```suggestion
    for (let i = 0; i <= maxIterations) {
        const { results: partialResults, url } = await api.get(url)
        results = results.concat(partialResults)
        if (!url) {
          break;
        }
    }  
```

Reply:
fair yeah - initially I wrote a while cause I didn't have the maxIterations thing, but great point